Xamarin.Forms.Core.UnitTests/BindableObjectUnitTests.cs

```diff
@@ -890,6 +890,23 @@ public void IsSetIsTrueWhenPropSet()
 			Assert.IsTrue(isSet);
 		}
 
+		[Test]
+		public void IsSetIsTrueWhenPropCleared()
+		{
+			string defaultValue = "default";
+			string newValue = "new value";
+
+			var bindableProperty = BindableProperty.Create("Foo", typeof(string), typeof(MockBindable), defaultValue);
+			var bindable = new MockBindable();
+
+			bindable.SetValue(bindableProperty, newValue);
+
+			bindable.ClearValue(bindableProperty);
+
+			var isSet = bindable.IsSet(bindableProperty);
+			Assert.IsTrue(isSet);
```

Comment:
I'll look at fixing this, or investigating why it is that way. if `ClearValue` is dropping the context, you could `ClearValue` the other property when the first is `!IsSet()` in your onPropertyChanged